plugins/ckan_operators/resource_operator.py

```diff
@@ -9,53 +9,186 @@
 
 class GetOrCreateResourceOperator(BaseOperator):
     """
-    Returns CKAN resource object, or creates it if it does not exist 
+    Returns CKAN resource object, or creates it if it does not exist
+
+    Expects as inputs:
+    - address: CKAN instance URL
+    - apikey: CKAN API key 
+    - package_name_or_id: the name or id of the package wherein we'll look for/make a resource
+    - resource_name: the name of the resource this operator will look for/make
+    - resource_id: the resource_id this operator will look for/make
+    - resource_attributes: the attributes that will be in a resource if we create it
+
+    Each of the above (except resource attributes, address and apikey) can be given with an actual value or with a reference to a task_id and task_key that returns the value
     """
 
     @apply_defaults
     def __init__(
         self,
         address: str,
         apikey: str,
-        package_name_or_id: str,
+
+        package_name_or_id: str = None,
+        package_name_or_id_task_id: str = None,
+        package_name_or_id_task_key: str = None,
+
         resource_name: str = None,
+        resource_name_task_id: str = None,
+        resource_name_task_key: str = None,
+
         resource_id: str = None,
+        resource_id_task_id: str = None,
+        resource_id_task_key: str = None,
+
+
         resource_attributes: str = None,
+        #resource_id_filepath = None,
+
         **kwargs,
     ) -> None:
         super().__init__(**kwargs)
-        self.package_id = package_name_or_id
-        self.resource_name = resource_name
-        self.resource_id = resource_id
+        self.package_name_or_id, self.package_name_or_id_task_id, self.package_name_or_id_task_key = package_name_or_id, package_name_or_id_task_id, package_name_or_id_task_key
+        self.resource_name, self.resource_name_task_id, self.resource_name_task_key = resource_name, resource_name_task_id, resource_name_task_key
+        self.resource_id, self.resource_id_task_id, self.resource_id_task_key = resource_id, resource_id_task_id, resource_id_task_key
```

Comment:
If I follow properly, the operator expects as inputs **either** one of these:

- resource_id
- resource_id_task_id AND resource_id_task_key

So, if the Resource ID specifically isn't passed, it gets it from another operator's output. Correct?

Couple observations:

1. Is the `resource_id_task_key` parameter for flexibility? Thinking that if the Operator is designed to always expect the CKAN resource dictionary, the key will always be the same.
1. If both `resource_id`, `resource_id_task_id AND resource_id_task_key` the latter takes precedence. Is that expected? If this is an issue we should add an `assert` at this stage to prevent this. Several ways of doing this, could be along the lines of:


```python
conditions = [
  resource_id and (not resource_id_task_id and not resource_id_task_key),
  resource_id_task_id and resource_id_task_key and not resource_id,
]

assert any(conditions), "Must pass either of: resource_id OR both resource_id_task_id and resource_id_task_key"

```

Reply:
Do we need one or the other? If we get both, it should work fine too, no?

I understand getting both would be ... odd ... but is it worth throwing an error?

Reply:
The key concern for me if we get both: which one should take precedence? plus, could it be a developer error? it isn't immediately clear to me right now, and if it isn't clear to a dev which one takes precedence I imagine confusion will arise around how the code works.

Reply:
Future PR